src/components/ItemFilter/ItemFilter.tsx

```diff
@@ -153,25 +111,38 @@ function ItemFilter(props: Props) {
         })
     }
 
-    let isLevelSelectDisabled = () => {
-        return itemFilter && itemFilter.enchantment && itemFilter.enchantment.name ? (itemFilter.enchantment.name.toLowerCase() === "none" || itemFilter.enchantment.name.toLowerCase() === "any") : false;
-    }
 
-    let enchantmentSelectList = enchantments.map(enchantment => {
-        return (
-            <option data-id={enchantment.id} key={enchantment.id} value={enchantment.name}>{enchantment.name}</option>
-        )
-    })
 
-    let reforgeSelectList = reforges.map(reforge => {
+    let filterSelectList = props?.filters ? props?.filters.filter(f => !selectedFilters.includes(f)).map(filter => {
         return (
-            <option data-id={reforge.id} key={reforge.id} value={reforge.name}>{reforge.name}</option>
+            <option data-id={filter} key={filter} value={filter}>{filter}</option>
         )
-    })
+    }) : ""
+
+    let onFilterChange = (filter?: ItemFilter) => {
+        console.log(filter);
+        console.log(Object.keys(filter as object));
+
+        var keys = Object.keys(filter as object);
+        if (keys.length > 0) {
+            var key = keys[0];
+            itemFilter![key] = filter![key];
```

Comment:
can you be sure itemFilter and filter exist?
check with a if

Reply:
it does exist